src/main/java/com/khutircraftubackend/product/ProductMapper.java

```diff
@@ -26,49 +31,105 @@ default String multipartFileToString(MultipartFile file) {
     }
 
     default void validateCreateRequest(ProductCreateRequest request) {
+
         if (request.name() == null || request.name().isEmpty()) {
             throw new IllegalArgumentException(ProductValidationMessages.NAME_NULL_OR_EMPTY);
         }
+
         if (request.description() == null) {
             throw new IllegalArgumentException(ProductValidationMessages.DESCRIPTION_NULL);
         }
+
         if (request.image() == null || request.image().isEmpty()) {
             throw new IllegalArgumentException(ProductValidationMessages.IMAGE_NULL_OR_EMPTY);
         }
+
         if (request.thumbnailImage() == null || request.thumbnailImage().isEmpty()) {
             throw new IllegalArgumentException(ProductValidationMessages.THUMBNAIL_IMAGE_NULL_OR_EMPTY);
         }
+
         if (request.available() == null) {
             throw new IllegalArgumentException(ProductValidationMessages.AVAILABLE_NULL);
         }
-        if (request.seller() == null) {
+
+        if (request.sellerId() == null) {
             throw new IllegalArgumentException(ProductValidationMessages.SELLER_NULL);
         }
+
+        if(request.categoryId() == null) {
+            throw new IllegalArgumentException(ProductValidationMessages.CATEGORY_NULL_OR_EMPTY);
+        }
     }
 
     default void validateUpdateRequest(ProductUpdateRequest request) {
+
```

Comment:
i think it's not the best idea to put validation into mapper class.
is it some common example? 
how do you feel about having a separate class for it? or perhaps some annotation?

Reply:
Moved to service.